### ISSUE TITLE
The imp module is deprecated.

### ISSUE THREAD

**Issue by fredrikeaa** [pdm-project/pdm#574]
- [x] I have searched the issue tracker and believe that this is not a duplicate.

**Make sure you run commands with `-v` flag before pasting the output.**

See [sitecustomize.py](https://github.com/pdm-project/pdm/blob/fc9d0bb4e7b801facde79f31ae6710404b7669d6/pdm/pep582/sitecustomize.py).

## Steps to reproduce

```
pip3 install paver
pip3 install pdm
eval $(pdm --pep582)
export PYTHONWARNINGS=error
mkdir test
cd test
paver
```

## Actual behavior

```
Error in sitecustomize; set PYTHONVERBOSE for traceback:
DeprecationWarning: the imp module is deprecated in favour of importlib; see the module's documentation for alternative uses
```

## Expected behavior

No output.

## Environment Information

```bash
# Paste the output of `pdm info && pdm info --env` below:
$ pdm info && pdm info --env
Error in sitecustomize; set PYTHONVERBOSE for traceback:
DeprecationWarning: the imp module is deprecated in favour of importlib; see the module's documentation for alternative uses
Traceback (most recent call last):
  File "/home/fredrik/.local/lib/python3.8/site-packages/pip/_vendor/packaging/version.py", line 49, in parse
    return Version(version)
  File "/home/fredrik/.local/lib/python3.8/site-packages/pip/_vendor/packaging/version.py", line 266, in __init__
    raise InvalidVersion(f"Invalid version: '{version}'")
pip._vendor.packaging.version.InvalidVersion: Invalid version: 'pdm'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/fredrik/.local/bin/pdm", line 5, in <module>
    from pdm.core import main
  File "/home/fredrik/.local/lib/python3.8/site-packages/pdm/__init__.py", line 5, in <module>
    from pdm.cli.commands.base import BaseCommand
  File "/home/fredrik/.local/lib/python3.8/site-packages/pdm/cli/commands/base.py", line 6, in <module>
    from pdm.cli.utils import PdmFormatter
  File "/home/fredrik/.local/lib/python3.8/site-packages/pdm/cli/utils.py", line 12, in <module>
    from pip._vendor.pkg_resources import Distribution
  File "/home/fredrik/.local/lib/python3.8/site-packages/pip/_vendor/pkg_resources/__init__.py", line 3252, in <module>
    def _initialize_master_working_set():
  File "/home/fredrik/.local/lib/python3.8/site-packages/pip/_vendor/pkg_resources/__init__.py", line 3235, in _call_aside
    f(*args, **kwargs)
  File "/home/fredrik/.local/lib/python3.8/site-packages/pip/_vendor/pkg_resources/__init__.py", line 3264, in _initialize_master_working_set
    working_set = WorkingSet._build_master()
  File "/home/fredrik/.local/lib/python3.8/site-packages/pip/_vendor/pkg_resources/__init__.py", line 574, in _build_master
    ws = cls()
  File "/home/fredrik/.local/lib/python3.8/site-packages/pip/_vendor/pkg_resources/__init__.py", line 567, in __init__
    self.add_entry(entry)
  File "/home/fredrik/.local/lib/python3.8/site-packages/pip/_vendor/pkg_resources/__init__.py", line 623, in add_entry
    for dist in find_distributions(entry, True):
  File "/home/fredrik/.local/lib/python3.8/site-packages/pip/_vendor/pkg_resources/__init__.py", line 2061, in find_on_path
    path_item_entries = _by_version_descending(filtered)
  File "/home/fredrik/.local/lib/python3.8/site-packages/pip/_vendor/pkg_resources/__init__.py", line 2034, in _by_version_descending
    return sorted(names, key=_by_version, reverse=True)
  File "/home/fredrik/.local/lib/python3.8/site-packages/pip/_vendor/pkg_resources/__init__.py", line 2032, in _by_version
    return [packaging.version.parse(part) for part in parts]
  File "/home/fredrik/.local/lib/python3.8/site-packages/pip/_vendor/pkg_resources/__init__.py", line 2032, in <listcomp>
    return [packaging.version.parse(part) for part in parts]
  File "/home/fredrik/.local/lib/python3.8/site-packages/pip/_vendor/packaging/version.py", line 51, in parse
    return LegacyVersion(version)
  File "/home/fredrik/.local/lib/python3.8/site-packages/pip/_vendor/packaging/version.py", line 111, in __init__
    warnings.warn(
DeprecationWarning: Creating a LegacyVersion has been deprecated and will be removed in the next major release


Selected lines from env:
PYTHONVERBOSE=
PYTHONPATH=/home/fredrik/.local/lib/python3.8/site-packages/pdm/pep582
PYTHONWARNINGS=error

```

**Comment by frostming** [comment]
As for now, `sitecustomize.py` must be py2 compatible so IMO it is tolerable. We can update it when PDM officially drops support of Python 2.7.

**Comment by frostming** [comment]
Fixed in f26cccafcb7da082e6c5b3feaeda88872324f67c, close this now.
